### PR TITLE
Pivx wallet  pivx: 3.2.0 -> 3.4.0 

### PR DESCRIPTION
**Pull request by wucke13**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by mmahut** [comment]
This is marked as broken on master, please see 254292858c922fa3bf8c65c75f0868e38c5fe737. If you update supports this, please rebase and drop the broken flag.

**Comment by wucke13** [comment]
Interesting. I was able to build and run the application from this patched derivation. I will look into this soon.

**Comment by wucke13** [comment]
I fixed this by manually speciying `openssl_1_0_2` as dep. Why didn't you do that in the first place in favor of marking the package as broken? @globin @mmahut
At least building is no possible, on master with the most recent version of pivx.

**Comment by globin** [comment]
I don't trust upstream who haven't updated from an openssl version that has been EOL for years and includes a number of security issues. That is quite an alarming sign of negligence on their part.

**Comment by mmahut** [comment]
@globin has a good point, that the trust in the product is questionable in case they are using EOL libraries, even more if it's a cryptocurrency, however I agree with @wucke13 that we should leave it up to the maintainer to fix any problems with the package and only when we fail to get in touch with the maintainer we should mark the package as broken.

**Comment by wucke13** [comment]
Thank you for the fast responses! Indeed, it depends on `openssl-1.0.x`, neither `openssl-1.1.x` nor any `libressl` will work. While there was some discussion about getting totally rid of `openssl` as a dependency, I wasn't able to quickly find proof of someone actually working on it, so `openssl-1.0.2` is going to stay a dep for some more time, most likely. 
From that standpoint, I'm considering this PR ready for merge, as it won't become any better in the near future. Do you have any other feedback left?

PS: I ~guess~ hope that `pivx` mostly relies on openssl code for their consensus algorithm, and not for actual SSL tunneling. In that case using an outdated version might not be such a big problem?

**Comment by mmahut** [comment]
Hard to say how they are using it, but I would maybe open a github issue with them with the request to update, at least in the future? One last rebuild before merging.

@GrahamcOfBorg build altcoints.pivx

**Comment by wucke13** [comment]
@mmahut You had a typo, on `t` too much :)

After doing a quick research, I have to oppose to @globin 's claim of 

> [an] openssl version that has been EOL for years

According to https://mta.openssl.org/pipermail/openssl-announce/2019-May/000151.html openssl-1.0.2 is still maintained, with having its most recent release just four months back in the past. To me this looks like a 'not so well though-through' basis to convince any pivx maintainer of this being an actual issue.

**Comment by mmahut** [comment]
@GrahamcOfBorg build altcoins.pivx

**Comment by wucke13** [comment]
Where do they claim that?
https://github.com/PIVX-Project/PIVX/blob/master/doc/build-unix.md only mentiones `openssl-1.0`
Anyway, why didn't the build succeed? Shall I rebase on master once more?

**Comment by mmahut** [comment]
Looks like they have stopped using that check yesteday as per https://github.com/PIVX-Project/PIVX/pull/991.

Please rebase as there is a conflict, given the file has been moved from altcoins to blockchains.

**Comment by wucke13** [comment]
Ok. The package is still not evaluating with `openssl`, so I'm staying with `openssl_1_0_2` so far. I rebased on master and added a commit for pivx 3.4.0, hopefully this will work. I will edit this post, once compilation succeeded.

**Comment by mmahut** [comment]
@GrahamcOfBorg build pivx

**Comment by wucke13** [comment]
@mmahut 
I'm done I guesss. I added a test, added `wrapQtAppsHook` to solve the `qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""` issue. Should build both `pivx` and `pivxd` just fine.

**Comment by wucke13** [comment]
@GrahamcOfBorg build pivx
@GrahamcOfBorg build pivxd

**Comment by Lassulus** [comment]
tested with nix-review, started pivx-qt and clicked around a little bit, everything looked fine

File: pkgs/applications/blockchains/pivx.nix [hunk 1]
```diff
@@ -50,12 +56,9 @@ stdenv.mkDerivation rec {
...
     platforms = platforms.unix;
-
+    # TODO
     # upstream doesn't support newer openssl versions
     # https://github.com/PIVX-Project/PIVX/issues/748
-    # "Your system is most probably using openssl 1.1 which is not the
-    # officialy supported version. Either use 1.0.1 or run again configure
-    # with the given option."
-    broken = true;
+    # openssl_1_0_2 should be replaced with opennsl ASAP
```
Comment (Lassulus) [inline pkgs/applications/blockchains/pivx.nix]:
```suggestion
    # openssl_1_0_2 should be replaced with openssl ASAP
```

**Comment by mmahut** [comment]
@GrahamcOfBorg eval

**Comment by wucke13** [comment]
@Lassulus Thank you! Can we backport this to `19.09` too?

**Comment by Lassulus** [comment]
Sure, can you open a backport PR?

**Pull request merged**